app/components/claim-user.js

```diff
@@ -0,0 +1,90 @@
+import Component from '@ember/component';
+import { inject as service } from '@ember/service';
+import { alias } from '@ember/object/computed';
+import Analytics from 'ember-osf/mixins/analytics';
+import { task } from 'ember-concurrency';
+import config from 'ember-get-config';
+import $ from 'jquery';
+import { validator, buildValidations }
+    from 'ember-cp-validations';
+
+const Validations = buildValidations({
+    email: {
+        validators: [
+            validator('presence', true),
+            validator('format', {
+                type: 'email',
+                regex: /^[-a-z0-9~!$%^&*_=+}{'?]+(\.[-a-z0-9~!$%^&*_=+}{'?]+)*@([a-z0-9_][-a-z0-9_]*(\.[-a-z0-9_]+)*\.(aero|arpa|biz|com|coop|edu|gov|info|int|mil|museum|name|net|org|pro|travel|mobi|[a-z][a-z])|([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}))(:[0-9]{1,5})?$/i,
+            }),
+        ],
+    },
+});
+
+/**
+ * @module ember-preprints
+ * @submodule components
+ */
+
+/**
+ * Confirm share preprint modal
+ *
+ * Requires user to confirm they wish to submit their preprint,
+ * thus making it public and searchable
+ *
+ * Sample usage:
+ * ```handlebars
+ * {{confirm-share-preprint
+ *  isOpen=showModalSharePreprint
+ *  shareButtonDisabled=shareButtonDisabled
+ *  savePreprint=(action 'savePreprint')
+ *  title=title
+ *  buttonLabel=buttonLabel
+ *}}
+ * ```
+ * @class confirm-share-preprint
+ */
+export default Component.extend(Validations, Analytics, {
+    tagName: 'li',
+    toast: service(),
+    i18n: service(),
+    showPopup: false,
+    email: null,
+    preprintId: null,
+    author: null,
+    isValid: alias('validations.isValid'),
+    actions: {
+        hidePopup() {
+            this.set('showPopup', false);
+        },
+        togglePopup() {
+            this.toggleProperty('showPopup');
+        },
+    },
+    claimUser: task(function* () {
```

Comment:
I thought about doing this on the `users` model, but I don't think it is quite right. Here what I think:
If we are to put it in the `users` model, that means we have to override the `createRecord()` method, as well as the `urlForCreateRecord()` method of the adapter. That is too much footprint for something that is going to be used once here. 
Also, upon claiming the user, the API returns 204 with empty content, instead of details of an `OSFUser` (because technically the OSFUser is not created until the email is confirmed). That means ember doesn't really get a `user` instance in return when we do `user.save()`, which COULD cause some more problems we need to fix by adding more logic to the `user` serializer as this is probably not JSONAPI-compliant anyway. To go down that rabbit hole does not worth the hassle in my opinion.

Reply:
Was bugging jams about this, slightly better on the model, I suggest asking him for clarification on best way to do that, doesn't need to be adapter changes, etc.
```
But, either way, I agree the code for this should go in the user model. This abstracts it away from the component so you could go with the POST to /claim now and make it simpler later.
```